romancal/skymatch/skystatistics.py

```diff
@@ -126,14 +126,14 @@ def calc_sky(self, data):
             in `skyvalue`.
 
         """
-        imstat = ImageStats(image=data.value, fields=self._fields, **(self._kwargs))
+        imstat = ImageStats(image=data, fields=self._fields, **(self._kwargs))
         stat = self._skystat(imstat)  # dict or scalar
 
         # re-attach units:
         if hasattr(stat, "__len__"):
-            self.skyval = {k: value * data.unit for k, value in stat.items()}
+            self.skyval = {k: value for k, value in stat.items()}
```

Comment:
This is no longer necessary, all it is doing is attaching the units that were stripped off.

Reply:
Thanks, William! I updated the code to remove that check.